guide_drafts/trait_derives.md

```diff
@@ -93,96 +104,177 @@ fn main() {
+34 |         .load::<User>(&db_conn)
+   |          ^^^^ the trait `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` is not implemented for `(i32,)`
    |
    = help: the following implementations were found:
-             <(A,) as diesel::types::FromSqlRow<(SA,), DB>>
-   = note: required because of the requirements on the impl of `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` for `diesel_demo_cli::models::User`
-   = note: required because of the requirements on the impl of `diesel::LoadQuery<_, diesel_demo_cli::models::User>` for `diesel_demo_cli::schema::users::table`
+             <(A,) as diesel::Queryable<(SA,), DB>>
+   = note: required because of the requirements on the impl of `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>` for `diesel_demo::models::User`
+   = note: required because of the requirements on the impl of `diesel::LoadQuery<_, diesel_demo::models::User>` for `diesel_demo::schema::users::table`
 ```
 
 Notice the compiler is indicating a trait is not implmented for calling the `.load()` method.
 When reading, take note of the values in the tuple[s].
-> `FromSqlRow<(Integer, Text, Text, Text, Text)>` not implemented for `(i32,`).
+
+> `diesel::Queryable<(diesel::types::Integer, diesel::types::Text, diesel::types::Text, diesel::types::Text), _>`
 
 
-`FromSqlRow` is trying to convert those 4 types into the types on our `User` model struct.
+`Queryable` is trying to convert those 4 types into the types on our `User` model struct.
 Our model struct has the three String fields commented out, so it doesn't know what those `Text` columns are supposed to be converted to.
 Remember, `Queryable` structs represent the exact columns, value, 
 and ordering of your query's returned result,
 which we are violating here.
-`FromSqlRow` is expecting a tuple that looks like `(i32, String, String, String)`,
+`Queryable` is expecting a tuple that looks like `(i32, String, String, String)`,
 but we currently only have a tuple consisting of `(i32,)`.
 We need to add those `String` columns back for our code to compile again.
 See Rust's [data type docs] if you're unfamiliar with tuples or their syntax.
 
 [data type docs]: https://doc.rust-lang.org/1.21.0/book/second-edition/ch03-02-data-types.html#grouping-values-into-tuples
 
-This trait bound is illustrated below in the "hand-written" derive example.
+## <a name="queryable_by_name"></a>QueryableByName
+As stated above, a `Queryable` struct must match the order 
+and types of the result of your `SQL` query.
+What about the case where we can't guarantee those bounds?
+Diesel's query builder is great,
+but it's API doesn't cover every possible query you can ever write. 
```

Comment:
I found this sentence a bit hard to parse. What do you think about restructuring it like: "`Queryable` is the trait you normally use with Diesel's query builder. If you're using `sql_query`, you will instead need to implement `QueryableByName`"